src/ckb/config.rs

```diff
@@ -148,7 +148,7 @@ serde_with::serde_conv!(
 );
 
 #[serde_as]
-#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+#[derive(Serialize, Deserialize, Debug, Clone, Default, Eq, PartialEq, Hash)]
```

Comment:
These `Eq` and `Hash` added in this file is not necessary.

Suggest a clean up:

```diff
diff --git a/src/ckb/config.rs b/src/ckb/config.rs
index 9ddded8..d0e318d 100644
--- a/src/ckb/config.rs
+++ b/src/ckb/config.rs
@@ -148,7 +148,7 @@ serde_with::serde_conv!(
 );
 
 #[serde_as]
-#[derive(Serialize, Deserialize, Debug, Clone, Default, Eq, PartialEq, Hash)]
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
 pub struct UdtScript {
     pub code_hash: H256,
     #[serde_as(as = "ScriptHashTypeWrapper")]
@@ -158,7 +158,7 @@ pub struct UdtScript {
 }
 
 #[serde_as]
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Hash)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 pub struct UdtCellDep {
     #[serde_as(as = "DepTypeWrapper")]
     pub dep_type: DepType,
@@ -166,7 +166,7 @@ pub struct UdtCellDep {
     pub index: u32,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq, Hash)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
 pub struct UdtArgInfo {
     pub name: String,
     pub script: UdtScript,
@@ -174,7 +174,7 @@ pub struct UdtArgInfo {
     pub cell_deps: Vec<UdtCellDep>,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq, Hash)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
 pub struct UdtCfgInfos(pub Vec<UdtArgInfo>);
 
 impl FromStr for UdtCfgInfos {
diff --git a/src/fiber/types.rs b/src/fiber/types.rs
index 16b08ec..a651290 100644
--- a/src/fiber/types.rs
+++ b/src/fiber/types.rs
@@ -1498,7 +1498,7 @@ impl TryFrom<molecule_fiber::AnnouncementSignatures> for AnnouncementSignatures
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Hash)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct NodeAnnouncement {
     // Signature to this message, may be empty the message is not signed yet.
     pub signature: Option<EcdsaSignature>,
@@ -2274,7 +2274,7 @@ impl TryFrom<molecule_gossip::GossipMessage> for GossipMessage {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Hash)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub enum BroadcastMessage {
     NodeAnnouncement(NodeAnnouncement),
     ChannelAnnouncement(ChannelAnnouncement),

```

Reply:
See https://github.com/nervosnetwork/fiber/pull/308#discussion_r1882457051